pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/text/LuceneTextIndexCreator.java

```diff
@@ -0,0 +1,149 @@
+   * dictionary, forward and inverted index, a text index is also created
+   * if text search is enabled on a column.
+   * @param column column name
+   * @param segmentIndexDir segment index directory
+   * @param commit true if the index should be committed (at the end after all documents have
+   *               been added), false if index should not be committed
+   * Note on commit:
+   *               Once {@link org.apache.pinot.core.segment.creator.impl.SegmentColumnarIndexCreator}
+   *               finishes indexing all documents/rows for the segment, we need to commit and close
+   *               the Lucene index which will internally persist the index on disk, do the necessary
+   *               resource cleanup etc. We commit during {@link InvertedIndexCreator#seal()}
+   *               and close during {@link InvertedIndexCreator#close()}.
+   *               This lucene index writer is used by both offline and realtime (both during
+   *               indexing in-memory MutableSegment and later during conversion to offline).
+   *               Since realtime segment conversion is again going to go through the offline
+   *               indexing path and will do everything (indexing, commit, close etc), there is
+   *               no need to commit the index from the realtime side. So when the realtime segment
+   *               is destroyed (which is after the realtime segment has been committed and converted
+   *               to offline), we close this lucene index writer to release resources but don't commit.
+   *               This is the reason to have commit flag part of the constructor.
+   */
+  public LuceneTextIndexCreator(String column, File segmentIndexDir, boolean commit) {
+    _textColumn = column;
+    try {
+      File indexFile = new File(segmentIndexDir.getPath() + "/" + _textColumn + LUCENE_TEXT_INDEX_FILE_EXTENSION);
+      _indexDirectory = FSDirectory.open(indexFile.toPath());
+      StandardAnalyzer standardAnalyzer = new StandardAnalyzer();
+      IndexWriterConfig indexWriterConfig = new IndexWriterConfig(standardAnalyzer);
+      indexWriterConfig.setRAMBufferSizeMB(LUCENE_INDEX_MAX_BUFFER_SIZE_MB);
+      indexWriterConfig.setCommitOnClose(commit);
+      _indexWriter = new IndexWriter(_indexDirectory, indexWriterConfig);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to instantiate Lucene text index creator. Error: " + e);
+    }
+  }
+
+  public IndexWriter getIndexWriter() {
+    return _indexWriter;
+  }
+
+  @Override
+  public void addDoc(Object document, int docIdCounter) {
+    Document docToIndex = new Document();
+    docToIndex.add(new TextField(_textColumn, document.toString(), Field.Store.NO));
+    docToIndex.add(new StoredField(LUCENE_INDEX_DOC_ID_COLUMN_NAME, docIdCounter));
+    try {
+      _indexWriter.addDocument(docToIndex);
+    } catch (Exception e) {
+      throw new RuntimeException("Failure while adding a new document to index. Error: " + e);
```

Comment:
please include name of the column in the exception. I suggest you log an error with the message, and not include the word "Error" in the exception string. Did you mean `+ e.getMessage()` instead of `+e`?

Reply:
Done